amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -612,6 +620,32 @@ void cleanupShardConsumers(Set<ShardInfo> assignedShards) {
         }
     }
 
+    private void startExecutorDiagnosticsDaemon() {
+        log.info("Starting executor diagnostics daemon.");
+
+        Thread diagnosticsThread = new Thread(new ThreadGroup("Diagnostics"), () -> {
+            while (true) {
+                ExecutorStateEvent executorStateEvent = new ExecutorStateEvent(executorService, leaseCoordinator);
+                executorStateEvent.accept(diagnosticEventHandler);
+                try {
+                    Thread.sleep(executorDiagnosticsDaemonSleepTimeMillis);
+                } catch (InterruptedException e) {
+                    log.error("Executor diagnostics thread interrupted", e);
+                }
+            }
+        });
+
+        diagnosticsThread.setDaemon(true);
+        diagnosticsThread.start();
```

Comment:
Why not run this in a ScheduledExecutorService? That way it can be made sure that it cleanly shuts down.

Reply:
Per discussion moved this functionality to runProcessLoop() method in Scheduler